src/spec.ts

```diff
@@ -337,9 +337,34 @@ export function normalizeOverlay(spec: UnitSpec, overlayWithPoint: boolean, over
 
 // TODO: add vl.spec.validate & move stuff from vl.validate to here
 
-export function fieldDefs(spec: ExtendedUnitSpec): FieldDef[] {
-  // TODO: refactor this once we have composition
-  return vlEncoding.fieldDefs(spec.encoding);
+/* Accumulate non-duplicate fieldDefs in a dictionary */
+function accumulate(dict: any, fieldDefs: FieldDef[]): any {
+  fieldDefs.forEach(function(fieldDef) {
+    let key = hash(fieldDef);
```

Comment:
What if we use the same field twice -- once as linear scale and once as log scale.  This seems to produce duplicates?